src/AbstractMetricsManager.h

```diff
@@ -214,6 +214,7 @@ class AbstractMetricsBucket
 
     virtual void to_json(json &j) const = 0;
     virtual void to_prometheus(std::stringstream &out, Metric::LabelMap add_labels = {}) const = 0;
+    virtual void update_top_metrics(size_t top_count) = 0;
```

Comment:
let's do `topn_count` across the board, so `update_topn_metrics(size_t topn_count)`